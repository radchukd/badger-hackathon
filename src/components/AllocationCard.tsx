import React from 'react';
import { Cell, Pie, PieChart, Tooltip } from 'recharts';

import {
  Box,
  Card,
  CardContent,
  List,
  ListItem,
  ListItemAvatar,
  ListItemText,
  Typography,
  useMediaQuery,
} from '@material-ui/core';

import useCardStyles from '../styles/cardStyles';
import useGlobalStyles from '../styles/globalStyles';
import { formatNumber } from '../utils/numberUtils';

export interface IAllocationCard {
  title: string;
  subtitle: string;
  data: Array<{ name: string; value: number }>;
}

const AllocationCard: React.FC<IAllocationCard> = ({ title, subtitle, data }) => {
  const classes = { ...useCardStyles(), ...useGlobalStyles() };
  const isTabletOrSmaller = useMediaQuery('(max-width: 1280px)');

  const COLORS = ['#E8C6A7', '#6F80BB', '#304DB3', '#52B330', '#000000'];

  const getAssetIcon = (name: string): JSX.Element => {
    if (name === 'Other') return <Box className={classes.avatarBig} />;

    return <img src={`/assets/${name}.png`} width="40" height="40" />;
  };

  return (
    <Card className={classes.cardRoot}>
      <CardContent className={classes.content}>
        <Typography className={classes.cardSubheading4}>{title}</Typography>
        <Typography className={classes.cardSubheading2} color="textSecondary">
          {subtitle}
        </Typography>
        <Box display="flex" flexDirection={isTabletOrSmaller ? 'column' : 'row'} alignItems="center">
          <Box>
            <PieChart width={240} height={240}>
              <Tooltip
                isAnimationActive={false}
                formatter={(value: number, name: string) => [formatNumber(value, 'percent'), name]}
              />
              <Pie data={data} cx="50%" cy="50%" stroke="none" fill="#8884d8" dataKey="value">
                {data.map((_entry, index) => (
                  <Cell key={`cell-${index}`} fill={COLORS[index % COLORS.length]} />
                ))}
              </Pie>
            </PieChart>
          </Box>
          <Box>
            <List>
              {data.map((entry) => (
                <ListItem key={entry.name} data-testid={`allocation-card-${entry.name}`}>
                  <ListItemAvatar>{getAssetIcon(entry.name)}</ListItemAvatar>
                  <ListItemText
                    primary={entry.name}
                    primaryTypographyProps={{ className: classes.cardSubheading4 }}
                    secondary={formatNumber(entry?.value, 'percent')}
                    secondaryTypographyProps={{ className: classes.cardSubheading2 }}
                  />
                </ListItem>
              ))}
            </List>
          </Box>
        </Box>
      </CardContent>
    </Card>
  );
};

export default AllocationCard;
