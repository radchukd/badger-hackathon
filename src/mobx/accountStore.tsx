import { action, extendObservable } from 'mobx';
import fetch from 'node-fetch';
import testAccount from '../components/__tests__/testAccount.json';

import { Account } from '../model/account.interface';
import { SettBalance } from '../model/sett-balance.interface';
import { SettVault } from '../model/sett-vault.interface';

export type AccountStoreType = {
  preload: boolean;
};

export class AccountStore {
  private baseUrl = 'https://api.badger.finance/v2';
  public account?: Account | null;
  public prices?: Record<string, number>;
  public vaults?: Array<SettVault>;
  public lastUpdate: Date | null = null;

  constructor({ preload }: AccountStoreType) {
    extendObservable(this, {
      account: this.account,
      prices: this.prices,
      vaults: this.vaults,
    });

    if (preload) {
      this.loadAccount();
      this.loadPrices();
      this.loadVaults();
      this.lastUpdate = new Date();
    }
  }

  loadAccount = action(async (): Promise<void> => {
    //const res = await fetch(`${this.baseUrl}/accounts/0x4e65175f05b4140a0747c29cce997cd4bb7190d4`);

    //if (res.ok) this.account = await res.json();
    //else this.account = null;
    this.account = testAccount;
  });

  loadPrices = action(async (): Promise<void> => {
    const res = await fetch(`${this.baseUrl}/prices`);

    if (res.ok) this.prices = await res.json();
    else this.prices = {};
  });

  loadVaults = action(async (): Promise<void> => {
    const res = await fetch(`${this.baseUrl}/setts`);

    if (res.ok) this.vaults = await res.json();
    else this.vaults = [];
  });

  get isLoading(): boolean {
    return (
      typeof this.account === 'undefined' || typeof this.prices === 'undefined' || typeof this.vaults === 'undefined'
    );
  }

  get roiPercentage(): number {
    if (!this.account?.value || !this.account?.earnedValue) return 0;

    return this.account?.earnedValue / this.account?.value;
  }

  get totalClaimable(): number {
    return this.account?.claimableBalances?.reduce((prev, cur) => prev + (cur?.balance ?? 0), 0) ?? 0;
  }

  get earnedBadger(): number {
    return this.account?.balances?.find((balance) => balance.asset === 'BADGER')?.earnedValue ?? 0;
  }

  assetValue = (balance: SettBalance): number => {
    return balance.balance ? balance.value / balance.balance : 0;
  };

  assetValueBTC = (balance: SettBalance): number => {
    if (!this.prices) return 0;

    const btcPrice = this.prices['0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599'];

    if (!btcPrice) return 0;

    return this.assetValue(balance) / btcPrice;
  };

  allocationDistribution = (type: 'balance' | 'strategy'): Array<{ name: string; value: number }> => {
    const totalAllocation = this.totalAllocation(true, type);

    if (!totalAllocation) return [];

    const top4 =
      this.account?.balances
        ?.slice()
        ?.sort((b1, b2) =>
          type === 'balance'
            ? Math.abs(b2.value) - Math.abs(b1.value)
            : Math.abs(b2.earnedValue) - Math.abs(b1.earnedValue),
        )
        ?.slice(0, 4)
        ?.map(({ asset, value, earnedValue }) => ({
          name: asset,
          value: Math.abs((type === 'balance' ? value : earnedValue) / totalAllocation),
        })) ?? [];
    const top4total = top4?.reduce((prev, cur) => prev + cur.value ?? 0, 0);
    const other = { name: 'Other', value: 1 - top4total };

    return [...top4, other];
  };

  allocationPerAsset = (value: number, type: 'balance' | 'strategy'): number => {
    const totalAllocation = this.totalAllocation(true, type);

    if (!totalAllocation) return 0;

    return Math.abs(value / totalAllocation);
  };

  totalAllocation = (abs: boolean, type: 'balance' | 'strategy'): number => {
    return (
      this.account?.balances?.reduce((prev, cur) => {
        const curValue = (type === 'balance' ? cur.value : cur.earnedValue) ?? 0;
        return prev + (abs ? Math.abs(curValue) : curValue);
      }, 0) ?? 0
    );
  };

  strategyROI = (asset: string): number => {
    if (!this.vaults) return 0;

    const vault = this.vaults.find((vault) => vault.asset === asset);

    if (!vault) return 0;

    return vault.apr;
  };
}

/*
declare let window: Window & { ethereum };

if (!window.ethereum) {
    console.log('Non-Ethereum browser detected.');
  }

  const addresses: Array<string> = await window.ethereum.request({
    method: 'eth_requestAccounts',
  });

  if (addresses.length < 0) {
    console.log('No account found.');
  }

  const web3 = new Web3(Web3.givenProvider);

  addresses.forEach(async (address) => {
    const res = await fetch(`${this.baseUrl}/accounts/${address}`);

    if (res.ok) {
      this.account = await res.json();
      throw new Error('Account is found.');
    }
  });
*/
