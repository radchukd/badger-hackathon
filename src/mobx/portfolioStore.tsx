import { action, makeAutoObservable } from 'mobx';

export type Vault = { id: number; value: string; tags: Array<string> };

export class PortfolioStore {
  public selectedVault?: Vault;
  public vaults: Array<Array<Vault>> = [
    [
      { id: 0, value: 'wBTC/Digg', tags: ['UNI'] },
      { id: 1, value: 'Badger/wBTC', tags: ['UNI'] },
    ],
    [
      { id: 2, value: 'Wrapped BTC/Digg', tags: ['SUSHI'] },
      { id: 3, value: 'Wrapped BTC/Badger', tags: ['SUSHI'] },
      { id: 4, value: 'Wrapped BTC/Wrapped Ether', tags: ['SUSHI'] },
    ],
    [
      { id: 5, value: 'crvRenWBTC', tags: ['CURVE'] },
      { id: 6, value: 'renBTC/wBTC/sBTC', tags: ['CURVE'] },
      { id: 7, value: 'tBTC/sBTCCrv LP', tags: ['CURVE'] },
      { id: 8, value: 'crvRenWBTC', tags: ['CURVE', 'HARVEST'] },
    ],
  ];
  public selectedVaults: Array<number> = this.vaults.flat().map((vault) => vault.id);

  constructor() {
    makeAutoObservable(this);
  }

  setSelectedVault = action((vault: string | undefined): void => {
    if (vault === undefined) {
      this.selectedVault = undefined;
    } else {
      const normalizedName = vault.replace('Sushiswap ', '').replace('Convex ', '');
      this.selectedVault = this.vaults.flat().find((v) => v.value === normalizedName);
    }
  });

  get areAllSelected(): boolean {
    return this.vaults.flat().length === this.selectedVaults.length;
  }

  isVaultSelected = (vault: Vault): boolean => {
    return this.selectedVaults.includes(vault.id);
  };

  onAllClick = action((): void => {
    if (this.areAllSelected) this.selectedVaults = [];
    else this.selectedVaults = this.vaults.flat().map((vault) => vault.id);
  });

  onVaultClick = action((vault: Vault): void => {
    this.selectedVaults = this.isVaultSelected(vault)
      ? this.selectedVaults.filter((selectedVaultId) => selectedVaultId !== vault.id)
      : [...this.selectedVaults, vault.id];
  });
}
