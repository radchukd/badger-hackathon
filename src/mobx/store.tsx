import { action, extendObservable } from 'mobx';
import { RouterStore } from 'mobx-router';
import fetch from 'node-fetch';
import { Account } from '../model/account.interface';
import { SettBalance } from '../model/sett-balance.interface';
import { SettVault } from '../model/sett-vault.interface';

//declare let window: Window & { ethereum };

export class RootStore {
  private baseUrl = 'https://api.badger.finance/v2';
  public router: RouterStore<RootStore>;
  public account?: Account;
  public prices?: Record<string, number>;
  public vaults?: SettVault[];

  constructor() {
    this.router = new RouterStore<RootStore>(this);

    extendObservable(this, {
      account: this.account,
      prices: this.prices,
      vaults: this.vaults,
    });

    this.loadAccount();
    this.loadPrices();
    this.loadVaults();
  }

  loadAccount = action(async (): Promise<void> => {
    /* if (!window.ethereum) {
      console.log('Non-Ethereum browser detected.');
    }

    const addresses: Array<string> = await window.ethereum.request({
      method: 'eth_requestAccounts',
    });

    if (addresses.length < 0) {
      console.log('No account found.');
    }

    const web3 = new Web3(Web3.givenProvider);

    addresses.forEach(async (address) => {
      const res = await fetch(`${this.baseUrl}/accounts/${address}`);

      if (res.ok) {
        this.account = await res.json();
        throw new Error('Account is found.');
      }
    }); */

    const res = await fetch(`${this.baseUrl}/accounts/0x4e65175f05b4140a0747c29cce997cd4bb7190d4`);
    if (res.ok) {
      this.account = await res.json();
    }
  });

  loadPrices = action(async (): Promise<void> => {
    const res = await fetch(`${this.baseUrl}/prices`);
    if (res.ok) {
      this.prices = (await res.json()) || [];
    }
  });

  loadVaults = action(async (): Promise<void> => {
    const res = await fetch(`${this.baseUrl}/setts`);
    if (res.ok) {
      this.vaults = (await res.json()) || [];
    }
  });

  get isLoading(): boolean {
    return (
      typeof this.account === 'undefined' || typeof this.prices === 'undefined' || typeof this.vaults === 'undefined'
    );
  }

  get roiPercentage(): number {
    if (!this.account?.value || !this.account?.earnedValue) return 0;
    return this.account?.earnedValue / this.account?.value;
  }

  get totalClaimable(): number {
    return this.account?.claimableBalances?.reduce((prev, cur) => prev + (cur?.balance ?? 0), 0) ?? 0;
  }

  get earnedBadger(): number {
    if (!this.prices) return 0;

    const badgerPrice = this.prices['0x3472A5A71965499acd81997a54BBA8D852C6E53d'];

    if (!badgerPrice || !this.account?.earnedValue) return 0;

    return this.account?.earnedValue / badgerPrice;
  }

  assetValue = (balance: SettBalance): number => {
    return balance.balance ? balance.value / balance.balance : 0;
  };

  assetValueBTC = (balance: SettBalance): number => {
    if (!this.prices) return 0;

    const btcPrice = this.prices['0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599'];

    if (!btcPrice) return 0;

    return this.assetValue(balance) / btcPrice;
  };

  allocationDistribution = (type: 'balance' | 'strategy'): Array<{ name: string; value: number }> => {
    const totalAllocation = this.totalAllocation(true, type);

    if (!totalAllocation) return [];

    const top4 =
      this.account?.balances
        ?.slice()
        ?.sort((b1, b2) =>
          type === 'balance'
            ? Math.abs(b2.value) - Math.abs(b1.value)
            : Math.abs(b2.earnedValue) - Math.abs(b1.earnedValue),
        )
        ?.slice(0, 4)
        ?.map(({ asset, value, earnedValue }) => ({
          name: asset,
          value: Math.abs((type === 'balance' ? value : earnedValue) / totalAllocation),
        })) ?? [];
    const top4total = top4?.reduce((prev, cur) => prev + cur.value ?? 0, 0);
    const other = { name: 'Other', value: 1 - top4total };
    return [...top4, other];
  };

  allocationPerAsset = (value: number, type: 'balance' | 'strategy'): number => {
    const totalAllocation = this.totalAllocation(true, type);

    if (!totalAllocation) return 0;

    return Math.abs(value / totalAllocation);
  };

  totalAllocation = (abs: boolean, type: 'balance' | 'strategy'): number => {
    return (
      this.account?.balances?.reduce((prev, cur) => {
        const curValue = (type === 'balance' ? cur.value : cur.earnedValue) ?? 0;
        return prev + (abs ? Math.abs(curValue) : curValue);
      }, 0) ?? 0
    );
  };

  strategyROI = (asset: string): number => {
    if (!this.vaults) return 0;

    const vault = this.vaults.find((vault) => vault.asset === asset);

    if (!vault) return 0;

    return vault.apr;
  };
}

const store = new RootStore();

export default store;
