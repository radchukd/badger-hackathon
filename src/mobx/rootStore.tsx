import { RouterStore } from 'mobx-router';
import React from 'react';

import { AccountStore } from './accountStore';
import { PortfolioStore } from './portfolioStore';

export class RootStore {
  public router: RouterStore<RootStore>;

  constructor() {
    this.router = new RouterStore<RootStore>(this);
  }
}

export const rootStore = new RootStore();

export const accountStore = new AccountStore({ preload: true });

export const portfolioStore = new PortfolioStore();

export interface IStoreContext {
  rootStore: RootStore;
  accountStore: AccountStore;
  portfolioStore: PortfolioStore;
}

export const StoreContext = React.createContext({} as IStoreContext);

export const StoreProvider = StoreContext.Provider;
