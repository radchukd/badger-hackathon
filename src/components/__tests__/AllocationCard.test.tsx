import '@testing-library/jest-dom';

import React from 'react';

import { render, screen } from '@testing-library/react';

import { AccountStore } from '../../mobx/accountStore';
import AllocationCard, { IAllocationCard } from '../AllocationCard';

describe('AllocationCard', () => {
  const accountStore = new AccountStore({ preload: false });
  const { loadAccount, allocationDistribution } = accountStore;

  let props: IAllocationCard;

  beforeAll(async () => {
    await loadAccount();

    props = {
      title: 'Asset Allocation',
      subtitle: 'Subtext',
      data: allocationDistribution('balance'),
    };
  });

  it('renders correctly', () => {
    const tree = render(
      <AllocationCard
        {...{
          ...props,
          data: [
            {
              name: 'DIGG',
              value: 0.6193641151985265,
            },
            {
              name: 'crvsBTC',
              value: 0.12364468850869027,
            },
            {
              name: 'SLP-DIGG-WBTC',
              value: 0.08224361422643128,
            },
            {
              name: 'crvtBTC',
              value: 0.06387563501650294,
            },
            {
              name: 'Other',
              value: 0.11087194704984904,
            },
          ],
        }}
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('displays black circle for other assets', () => {
    render(<AllocationCard {...props} />);

    const otherListItem = screen.getByTestId('allocation-card-Other');
    const otherListItemAvatar = otherListItem.firstChild;
    expect(otherListItemAvatar?.firstChild).toHaveStyle({ backgroundColor: 'rgb(0, 0, 0)' });
  });
});
