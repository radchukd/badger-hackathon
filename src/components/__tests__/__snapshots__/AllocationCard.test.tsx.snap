// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllocationCard renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-cardRoot-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root makeStyles-content-2"
        >
          <p
            class="MuiTypography-root makeStyles-cardSubheading4-8 MuiTypography-body1"
          >
            Asset Allocation
          </p>
          <p
            class="MuiTypography-root makeStyles-cardSubheading2-6 MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            Subtext
          </p>
          <div
            class="MuiBox-root MuiBox-root-15"
          >
            <div
              class="MuiBox-root MuiBox-root-16"
            >
              <div
                class="recharts-wrapper"
                style="position: relative; cursor: default; width: 240px; height: 240px;"
              >
                <svg
                  class="recharts-surface"
                  cx="50%"
                  cy="50%"
                  height="240"
                  version="1.1"
                  viewBox="0 0 240 240"
                  width="240"
                >
                  <defs>
                    <clippath
                      id="recharts1-clip"
                    >
                      <rect
                        height="230"
                        width="230"
                        x="5"
                        y="5"
                      />
                    </clippath>
                  </defs>
                  <g
                    class="recharts-layer recharts-pie"
                  >
                    <g
                      class="recharts-layer"
                    >
                      <g
                        class="recharts-layer recharts-pie-sector"
                      />
                      <g
                        class="recharts-layer recharts-pie-sector"
                      />
                      <g
                        class="recharts-layer recharts-pie-sector"
                      />
                      <g
                        class="recharts-layer recharts-pie-sector"
                      />
                      <g
                        class="recharts-layer recharts-pie-sector"
                      />
                    </g>
                  </g>
                </svg>
                <div
                  class="recharts-tooltip-wrapper"
                  style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
                >
                  <div
                    class="recharts-default-tooltip"
                    style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
                  >
                    <p
                      class="recharts-tooltip-label"
                      style="margin: 0px;"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root MuiBox-root-17"
            >
              <ul
                class="MuiList-root MuiList-padding"
              >
                <li
                  class="MuiListItem-root MuiListItem-gutters"
                  data-testid="allocation-card-DIGG"
                >
                  <div
                    class="MuiListItemAvatar-root"
                  >
                    <img
                      height="40"
                      src="/assets/DIGG.png"
                      width="40"
                    />
                  </div>
                  <div
                    class="MuiListItemText-root MuiListItemText-multiline"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardSubheading4-8 MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      DIGG
                    </span>
                    <p
                      class="MuiTypography-root makeStyles-cardSubheading2-6 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                    >
                      61.94%
                    </p>
                  </div>
                </li>
                <li
                  class="MuiListItem-root MuiListItem-gutters"
                  data-testid="allocation-card-crvsBTC"
                >
                  <div
                    class="MuiListItemAvatar-root"
                  >
                    <img
                      height="40"
                      src="/assets/crvsBTC.png"
                      width="40"
                    />
                  </div>
                  <div
                    class="MuiListItemText-root MuiListItemText-multiline"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardSubheading4-8 MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      crvsBTC
                    </span>
                    <p
                      class="MuiTypography-root makeStyles-cardSubheading2-6 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                    >
                      12.36%
                    </p>
                  </div>
                </li>
                <li
                  class="MuiListItem-root MuiListItem-gutters"
                  data-testid="allocation-card-SLP-DIGG-WBTC"
                >
                  <div
                    class="MuiListItemAvatar-root"
                  >
                    <img
                      height="40"
                      src="/assets/SLP-DIGG-WBTC.png"
                      width="40"
                    />
                  </div>
                  <div
                    class="MuiListItemText-root MuiListItemText-multiline"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardSubheading4-8 MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      SLP-DIGG-WBTC
                    </span>
                    <p
                      class="MuiTypography-root makeStyles-cardSubheading2-6 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                    >
                      8.22%
                    </p>
                  </div>
                </li>
                <li
                  class="MuiListItem-root MuiListItem-gutters"
                  data-testid="allocation-card-crvtBTC"
                >
                  <div
                    class="MuiListItemAvatar-root"
                  >
                    <img
                      height="40"
                      src="/assets/crvtBTC.png"
                      width="40"
                    />
                  </div>
                  <div
                    class="MuiListItemText-root MuiListItemText-multiline"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardSubheading4-8 MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      crvtBTC
                    </span>
                    <p
                      class="MuiTypography-root makeStyles-cardSubheading2-6 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                    >
                      6.39%
                    </p>
                  </div>
                </li>
                <li
                  class="MuiListItem-root MuiListItem-gutters"
                  data-testid="allocation-card-Other"
                >
                  <div
                    class="MuiListItemAvatar-root"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-18 makeStyles-avatarBig-14"
                    />
                  </div>
                  <div
                    class="MuiListItemText-root MuiListItemText-multiline"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardSubheading4-8 MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      Other
                    </span>
                    <p
                      class="MuiTypography-root makeStyles-cardSubheading2-6 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                    >
                      11.09%
                    </p>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-cardRoot-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root makeStyles-content-2"
      >
        <p
          class="MuiTypography-root makeStyles-cardSubheading4-8 MuiTypography-body1"
        >
          Asset Allocation
        </p>
        <p
          class="MuiTypography-root makeStyles-cardSubheading2-6 MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          Subtext
        </p>
        <div
          class="MuiBox-root MuiBox-root-15"
        >
          <div
            class="MuiBox-root MuiBox-root-16"
          >
            <div
              class="recharts-wrapper"
              style="position: relative; cursor: default; width: 240px; height: 240px;"
            >
              <svg
                class="recharts-surface"
                cx="50%"
                cy="50%"
                height="240"
                version="1.1"
                viewBox="0 0 240 240"
                width="240"
              >
                <defs>
                  <clippath
                    id="recharts1-clip"
                  >
                    <rect
                      height="230"
                      width="230"
                      x="5"
                      y="5"
                    />
                  </clippath>
                </defs>
                <g
                  class="recharts-layer recharts-pie"
                >
                  <g
                    class="recharts-layer"
                  >
                    <g
                      class="recharts-layer recharts-pie-sector"
                    />
                    <g
                      class="recharts-layer recharts-pie-sector"
                    />
                    <g
                      class="recharts-layer recharts-pie-sector"
                    />
                    <g
                      class="recharts-layer recharts-pie-sector"
                    />
                    <g
                      class="recharts-layer recharts-pie-sector"
                    />
                  </g>
                </g>
              </svg>
              <div
                class="recharts-tooltip-wrapper"
                style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
              >
                <div
                  class="recharts-default-tooltip"
                  style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
                >
                  <p
                    class="recharts-tooltip-label"
                    style="margin: 0px;"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-17"
          >
            <ul
              class="MuiList-root MuiList-padding"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters"
                data-testid="allocation-card-DIGG"
              >
                <div
                  class="MuiListItemAvatar-root"
                >
                  <img
                    height="40"
                    src="/assets/DIGG.png"
                    width="40"
                  />
                </div>
                <div
                  class="MuiListItemText-root MuiListItemText-multiline"
                >
                  <span
                    class="MuiTypography-root makeStyles-cardSubheading4-8 MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    DIGG
                  </span>
                  <p
                    class="MuiTypography-root makeStyles-cardSubheading2-6 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    61.94%
                  </p>
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters"
                data-testid="allocation-card-crvsBTC"
              >
                <div
                  class="MuiListItemAvatar-root"
                >
                  <img
                    height="40"
                    src="/assets/crvsBTC.png"
                    width="40"
                  />
                </div>
                <div
                  class="MuiListItemText-root MuiListItemText-multiline"
                >
                  <span
                    class="MuiTypography-root makeStyles-cardSubheading4-8 MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    crvsBTC
                  </span>
                  <p
                    class="MuiTypography-root makeStyles-cardSubheading2-6 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    12.36%
                  </p>
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters"
                data-testid="allocation-card-SLP-DIGG-WBTC"
              >
                <div
                  class="MuiListItemAvatar-root"
                >
                  <img
                    height="40"
                    src="/assets/SLP-DIGG-WBTC.png"
                    width="40"
                  />
                </div>
                <div
                  class="MuiListItemText-root MuiListItemText-multiline"
                >
                  <span
                    class="MuiTypography-root makeStyles-cardSubheading4-8 MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    SLP-DIGG-WBTC
                  </span>
                  <p
                    class="MuiTypography-root makeStyles-cardSubheading2-6 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    8.22%
                  </p>
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters"
                data-testid="allocation-card-crvtBTC"
              >
                <div
                  class="MuiListItemAvatar-root"
                >
                  <img
                    height="40"
                    src="/assets/crvtBTC.png"
                    width="40"
                  />
                </div>
                <div
                  class="MuiListItemText-root MuiListItemText-multiline"
                >
                  <span
                    class="MuiTypography-root makeStyles-cardSubheading4-8 MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    crvtBTC
                  </span>
                  <p
                    class="MuiTypography-root makeStyles-cardSubheading2-6 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    6.39%
                  </p>
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters"
                data-testid="allocation-card-Other"
              >
                <div
                  class="MuiListItemAvatar-root"
                >
                  <div
                    class="MuiBox-root MuiBox-root-18 makeStyles-avatarBig-14"
                  />
                </div>
                <div
                  class="MuiListItemText-root MuiListItemText-multiline"
                >
                  <span
                    class="MuiTypography-root makeStyles-cardSubheading4-8 MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    Other
                  </span>
                  <p
                    class="MuiTypography-root makeStyles-cardSubheading2-6 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    11.09%
                  </p>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
